torch/distributed/_functional_collectives.py

```diff
@@ -353,6 +353,60 @@ def all_to_all_single(
     return _maybe_wrap_tensor(tensor)
 
 
+def batch_isend_irecv(p2p_op_list: List[c10d.P2POp]) -> List[torch.Tensor]:
+    """
+    Send or Receive a batch of tensors asynchronously and return a list of requests.
+
+    This is the functional implementation of :func:`~torch.distributed.batch_isend_irecv`.
+
+    .. warning:: Gloo backend is not supported.
+
+    Args:
+        p2p_op_list: A list of point-to-point operations(type of each operator is
+            ``torch.distributed.P2POp``). The order of the isend/irecv in the list
+            matters and it needs to match with corresponding isend/irecv on the
+            remote end.
+
+    Returns:
+        A list of Received Tensors
```

Comment:
Would super appreciate some feedback on the return type here! Initially this was written such that the output contained all tensors in the p2p_op_list, but this requires us to clone the `send` tensor to satisfy functional requirements. 

One major drawback of not returning the send tensors is it also requires us to drop gloo support. Gloo returns a work object for each p2p_op, so without returning a send object there is nowhere to store the wait.

Reply:
i guess you didn't try fx tracing this yet? i am surprised if we can (or want to) represent the p2pOp in the fx graph.

probably makes sense to have a list of pure tensors?  im not sure how to deal with mixtures of sends/recvs though.